internal/pagination/pagination.go

```diff
@@ -0,0 +1,53 @@
+package pagination
```

Comment:
That is awfully long, I suggest `page` or something shorter: https://go.dev/blog/package-names